ctags/main/lregex.c

```diff
@@ -113,11 +107,11 @@ static void clearPatternSet (const langType language)
 			{
 				eFree (p->u.tag.name_pattern);
 				p->u.tag.name_pattern = NULL;
-				eFree (p->u.tag.kind.name);
+				eFree ((char *)p->u.tag.kind.name);
```

Comment:
it's not worse than [UCTags'](https://github.com/universal-ctags/ctags/blob/master/main/lregex.c#L316) despite what you say in the commit message.